macos/video-toolbox/src/video_encoder.rs

```diff
@@ -300,15 +305,57 @@ mod test {
         let mut encoder = VideoEncoder::new(VideoEncoderConfig {
             width: 1920,
             height: 1080,
-            fps: 29.97,
+            fps: 30.0,
             codec,
             input_format: VideoEncoderInputFormat::Yuv420Planar,
+            max_key_frame_interval: None,
         })
         .unwrap();
 
         let u = vec![200u8; 1920 * 1080 / 4];
         let v = vec![128u8; 1920 * 1080 / 4];
-        for i in 0..90 {
+        let mut keyframe_count = 0;
+        for i in 0..360 {
+            let mut y = Vec::with_capacity(1920 * 1080);
+            for line in 0..1080 {
+                let sample = if line / 12 == i {
+                    // add some motion by drawing a line that moves from top to bottom
+                    16
+                } else {
+                    (16.0 + (line as f64 / 1080.0) * 219.0).round() as u8
+                };
+                y.resize(y.len() + 1920, sample);
+            }
+            let frame = TestFrame {
+                samples: vec![y, u.clone(), v.clone()],
+            };
+            if i % 90 == 0 {
+                if let Some(output) = encoder.encode(frame, EncodedFrameType::Key).unwrap() {
+                    keyframe_count += if output.encoded_frame.is_keyframe { 1 } else { 0 };
+                }
+            } else if let Some(output) = encoder.encode(frame, EncodedFrameType::Auto).unwrap() {
+                keyframe_count += if output.encoded_frame.is_keyframe { 1 } else { 0 };
+            }
```

Comment:
Could be a little more DRY:

```suggestion
            if let Some(output) = encoder.encode(frame, if i % 90 == 0 { EncodedFrameType::Key } else { EncodedFrameType::Key }).unwrap() {
                keyframe_count += if output.encoded_frame.is_keyframe { 1 } else { 0 };
            }
```

Reply:
Ah, very neat! Thanks.